File: cpp/66d_1895B_PointsAndMinimummDistance.cpp
#include<bits/stdc++.h>

using namespace std;

int get_distance(vector<int> a){
    int cnt = 0;
    for(int i = 1; i < (a.size() / 2); i++){
        cnt += (a[i] - a[i-1]);
    }
    for(int i = (a.size() / 2) + 1; i < a.size(); i++){
        cnt += (a[i] - a[i-1]);;
    }
    return cnt;
}

int main(){
    ios::sync_with_stdio(0);
    cin.tie(0);

    int t, n, temp;
    vector<vector<int>> arr;
    cin>>t;

    for(int i = 0; i < t; i++){
        cin>>n;
        vector<int> a;
        for (int j = 0; j < (2 * n); j++){
            cin>>temp;
            a.push_back(temp);
        }
        arr.push_back(a);
    }

    for(int i = 0; i < t; i++){
        sort(arr[i].begin(), arr[i].end());
        cout<<get_distance(arr[i])<<"\n";
        for(int j = 0; j < (arr[i].size() / 2); j++){
            cout<<arr[i][j]<<" "<<arr[i][(arr[i].size() / 2)+ j]<<"\n";
        }
        cout<<"\n";
    }
    return 0;
}

File: python/66d_1895B_PointsAndMinimummDistance.py
def get_dist(a):
    cnt = 0
    for i in range(1, len(a) // 2):
        cnt += (a[i] - a[i-1])
    for i in range((len(a) // 2 )+1, len(a)):
        cnt += (a[i] - a[i-1])
    return cnt

def main():
    arr = []
    t = int(input())
    for _ in range(t):
        n = int(input())
        a = list(map(int, input().split()))
        arr.append(a)
    for i in arr:
        i.sort()
        print(get_dist(i))
        for j in range(len(i) // 2):
            print(i[j], i[(len(i) // 2) + j])

if __name__ == "__main__":
    main()